src/reynir_correct/pattern.py

```diff
@@ -1816,14 +1816,29 @@ def né(self, match: SimpleTree) -> None:
         if c is None:
             return
         start, end = c.span[0], c.span[1]
+        if (
```

Comment:
This looks suspicious. Why would you need to remove a previous annotation, isn't the same test applied to that one so it should never be added in the first place?

Reply:
In short, to answer the last three comments, P_NT_Né is from the error grammar rule VillaNé, which is fairly open. P_Né is from searching the syntax tree afterwards, capturing more cases. This is done after, so we might have already picked up the error with the grammar rule, but here we check more conditions, since it is easier here. After looking at examples, we now annotate all cases of 'né' except when we find 'hvorki'. It is highly unlikely that a sentence contains two instances of 'hvorki X né Y', so this should cover almost all cases, and seems to perform well. We also skip 'eitt né neitt', as that is (I think) the only valid phrase using 'né' without 'hvorki'.

Reply:
After making the P_Né function more robust I'm wondering if the VillaNé error grammar rule is needed anymore. Should I delete it?

Reply:
After looking at the examples, it seems a couple of sentences can no longer be parsed if VillaNé is not a part of the grammar. My suggestion is to include it in the main grammar rule set for now; I've made those changes in the parsefixes pull request on GreynirPackage.